microsite documentation

patofp opened the pull request:
# Description

Added documentation about reasoning engine and updated old documentation.
Added drawio diagrams automatic generation.

## Type of change

Make sure you add the correct label for the PR:

- `enhacement` if this PR adds a new feature
- `bug` if it fixes a bug
- `documentation` if it adds docs
- `breaking-change` if this PR introduces a breaking change
- `dependency-updates` if it updates dependencies

<!-- Does this PR fix any issue? add `CLOSES #numberofissue` under this line -->

L-Lavigne commented on build.sbt:
```diff
@@ -103,23 +119,36 @@ lazy val `bellman-spark-engine` = project
+      val `spark-testing-base` =
+        `spark`(scalaVersion.value) + "_" + `spark-testing`(scalaVersion.value)
+      Seq(
+        "org.apache.spark" %% "spark-sql" % `spark`(
```
@patofp Can we exclude sbt files from scalafix? It's being over-zealous with breaking up "long" lines, and messing up all our formatting.

patofp replied:
I think it is scalafmt the one that is doing the formatting, I would say this rules are causing the formatting:

```
style = defaultWithAlign
maxColumn = 80
```

We can exclude them by doing something like: 

```
project.excludeFilters = [".*\\.sbt"]
```

I will give a try to it. 

Said that, IMHO I suggest don't performing this change because although it could be annoying some times (like this one) we will also loose all of the formatting features that provide scalafmt. 

Also in a more profound way, I think that the greater automation and constraints in formatting rules, the less freedom to the human developer of performing a format error, thus less bureaucracy of creating, maintaining and applying a style guide by the developers and consequently less effort at PRs reviews, as the developer don't have to worry about style as far as possible.

patofp replied:
By the way I've noticed some formatting issues in this PR: https://github.com/gsk-aiops/bellman/pull/677/files
Not sure what is the cause of this. 🤔

L-Lavigne replied:
I fully agree about using scalafmt on our Scala files to enforce a consistent style. But I think it leads to bad results on sbt files specifically. For example it's common practice for developers to align sbt dependency declarations by columns: compare these two dependency blocks, [manually formatted](https://github.com/gsk-aiops/bellman/blob/0cd744c189cd36b65d4c8aa4c5c878daa3eb7765/build.sbt#L106) vs [scalafmt-formatted](https://github.com/gsk-aiops/bellman/blob/85f909e2c1790d66f65927c694cc2b234490fe55/build.sbt#L123). It seems clear to me that the manual formatting is clearer in this case.

We could change the alignment rule and increase the max line length, but that would affect Scala files as well which we agree is unwanted. So I think excluding sbt specifically as you described would be a good compromise. Please let me know your thoughts on this, thanks!

patofp replied:
>> "...compare these two dependency blocks, [manually formatted](https://github.com/gsk-aiops/bellman/blob/0cd744c189cd36b65d4c8aa4c5c878daa3eb7765/build.sbt#L106) vs [scalafmt-formatted](https://github.com/gsk-aiops/bellman/blob/85f909e2c1790d66f65927c694cc2b234490fe55/build.sbt#L123)."

Indeed, it looks way more clearer. I have to investigate what happened because I think we have some formatting issue. Since this PR, that is autogenerated, `build.sbt` it is being reformatted:  https://github.com/gsk-aiops/bellman/pull/677/files

patofp replied:
> > > "...compare these two dependency blocks, [manually formatted](https://github.com/gsk-aiops/bellman/blob/0cd744c189cd36b65d4c8aa4c5c878daa3eb7765/build.sbt#L106) vs [scalafmt-formatted](https://github.com/gsk-aiops/bellman/blob/85f909e2c1790d66f65927c694cc2b234490fe55/build.sbt#L123)."
> 
> Indeed, it looks way more clearer. I have to investigate what happened because I think we have some formatting issue. Since this PR, that is autogenerated, `build.sbt` it is being reformatted: https://github.com/gsk-aiops/bellman/pull/677/files



> > > "...compare these two dependency blocks, [manually formatted](https://github.com/gsk-aiops/bellman/blob/0cd744c189cd36b65d4c8aa4c5c878daa3eb7765/build.sbt#L106) vs [scalafmt-formatted](https://github.com/gsk-aiops/bellman/blob/85f909e2c1790d66f65927c694cc2b234490fe55/build.sbt#L123)."
> 
> Indeed, it looks way more clearer. I have to investigate what happened because I think we have some formatting issue. Since this PR, that is autogenerated, `build.sbt` it is being reformatted: https://github.com/gsk-aiops/bellman/pull/677/files

Ok I saw that the reformatting was been done since the Spark mirgration PR as it was manually formatted right? If thats the case then  the added rule: `project.excludeFilters = [".*\\.sbt"]` should work on CI from now on. 👍

patofp replied:
> > > "...compare these two dependency blocks, [manually formatted](https://github.com/gsk-aiops/bellman/blob/0cd744c189cd36b65d4c8aa4c5c878daa3eb7765/build.sbt#L106) vs [scalafmt-formatted](https://github.com/gsk-aiops/bellman/blob/85f909e2c1790d66f65927c694cc2b234490fe55/build.sbt#L123)."
> 
> Indeed, it looks way more clearer. I have to investigate what happened because I think we have some formatting issue. Since this PR, that is autogenerated, `build.sbt` it is being reformatted: https://github.com/gsk-aiops/bellman/pull/677/files



> > > "...compare these two dependency blocks, [manually formatted](https://github.com/gsk-aiops/bellman/blob/0cd744c189cd36b65d4c8aa4c5c878daa3eb7765/build.sbt#L106) vs [scalafmt-formatted](https://github.com/gsk-aiops/bellman/blob/85f909e2c1790d66f65927c694cc2b234490fe55/build.sbt#L123)."
> 
> Indeed, it looks way more clearer. I have to investigate what happened because I think we have some formatting issue. Since this PR, that is autogenerated, `build.sbt` it is being reformatted: https://github.com/gsk-aiops/bellman/pull/677/files

Ok I saw that the reformatting was been done since the Spark mirgration PR as it was manually formatted right? If thats the case then  the added rule: `project.excludeFilters = [".*\\.sbt"]` should work on CI from now on. 👍

patofp commented:
> # Description
> Added documentation about reasoning engine and updated old documentation. Added drawio diagrams automatic generation.
> 
> ## Type of change
> Make sure you add the correct label for the PR:
> 
> * `enhacement` if this PR adds a new feature
> * `bug` if it fixes a bug
> * `documentation` if it adds docs
> * `breaking-change` if this PR introduces a breaking change
> * `dependency-updates` if it updates dependencies

L-Lavigne commented on build.sbt:
```diff
@@ -247,8 +171,10 @@ lazy val `bellman-site` = project
-      "org.apache.spark" %% "spark-sql" % `spark`(scalaVersion.value) % Provided,
-      "io.github.stanch" %% "reftree"   % "1.4.0",
+      "org.apache.spark" %% "spark-sql"   %       `spark`(scalaVersion.value)  % Provided,
+      // TODO: Update, io.github.stanch.reftree 1.4.0 depends on com.lihaoyi.fastparse 2.1.2
```
Would it help if we downgraded fastparse in Bellman?
